types/src/v2/params.rs

```diff
@@ -482,4 +506,26 @@ mod test {
 		assert_eq!(dsr.subscription, SubscriptionId::Str("9".into()));
 		assert_eq!(dsr.result, serde_json::json!("offside"));
 	}
+
+	#[test]
+	fn params_sequence_optional_ignore_empty() {
+		let params = RpcParams::new(Some(r#"["foo", "bar"]"#));
+		let mut seq = params.sequence();
+
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), Some("foo"));
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), Some("bar"));
+
+		let params = RpcParams::new(Some(r#"[]"#));
+		let mut seq = params.sequence();
+		assert_eq!(seq.optional_next_ignore_empty_json::<&str>().unwrap(), None);
+
+		let params = RpcParams::new(Some(r#"[12, "[]"]"#));
```

Comment:
Curious about why you added `"` around the empty element here. Does this work the same without it? I.e. `r#"[12, []]"#`

Reply:
good point, it was mostly that I was concerned about a edge case, empty array vs empty element.

cool thanks, we should test both.